src/reconcile/rec_visibility.c

```diff
@@ -1044,14 +1054,17 @@ __rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_CELL_UNPACK_KV *
          * with read uncommitted isolation and we may see a committed update followed by uncommitted
          * updates
          */
-        if (!F_ISSET(r, WT_REC_EVICT) || !WT_IS_METADATA(session->dhandle))
+        if (!is_inmem_btree && (!F_ISSET(r, WT_REC_EVICT) || !WT_IS_METADATA(session->dhandle)))
             break;
     }
 
     /* The prepare rollback is stable. Delete the key by selecting the rollback tombstone. */
     if (upd_select->upd == NULL && prepare_rollback_tombstone != NULL)
         upd_select->upd = prepare_rollback_tombstone;
-
+    if (is_inmem_btree && oldest_visible_upd != NULL) {
```

Comment:
This is only correct if we don't have an on-page value. If we have an on-page value, we need to decide if the oldest value is globally visible, otherwise we should write nothing and keep the current on-disk value. We should also set the has_newer_update flag here if we have skipped any update.

Reply:
thank you, updated. Can you check if the logic is correct please?